(pre-review) [framework] introduce private entry function for creating multisig account

yeptos opened the pull request:
This eliminates the need for generating the complex proof the existing create_with_existing_account functions require.

## Description
<!-- Please include a summary of the change, including which issue it fixes or what feature it adds. Include relevant motivation, context and documentation as appropriate. List dependencies that are required for this change, if any. -->

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
<!--
- Please ensure that the functionality introduced by this change is well tested and verified to work as expected.
- Ensure tests cover both happy and unhappy paths.
- List and link relevant tests.
-->

## Key Areas to Review
<!--
- Identify any critical parts of the code that require special attention or understanding. Explain why these parts are crucial to the functionality or architecture of the project.
- Point out any areas where complex logic has been implemented. Provide a brief explanation of the logic and your approach to make it easier for reviewers to follow.
- Highlight any areas where you are particularly concerned or unsure about the code's impact on the change. This can include potential performance or security issues, or compatibility with existing features.
-->

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

<!-- Thank you for your contribution! -->

gregnazario committed:
[framework] Disable object burn

JoshLind committed:
[Consensus Observer] Downgrade unnecessary error logs.

JoshLind committed:
[Consensus Observer] Wrap block and payload stores in locks.

JoshLind committed:
[Consensus Observer] Handle duplicate commit messages.

JoshLind committed:
[Consensus Observer] Handle duplicate payload messages.

JoshLind committed:
[Consensus Observer] Handle duplicate ordered block messages.

JoshLind committed:
[Consensus Observer] Small renames and refactors.

JoshLind committed:
[Consensus Observer] Support multiple subscriptions.

yeptos committed:
[framework] introduce private entry function for creating multisig account
This eliminates the need for generating the complex proof the existing
create_with_existing_account functions require.

github-actions[bot] committed:
Update Docker images (#14565)

Co-authored-by: gedigi <19227040+gedigi@users.noreply.github.com>

rahxephon89 committed:
[Prover] Bitwise operation in aborts_if condition (#14552)

* temp

* update CI action

0xjunha committed:
Large Package Publishing module with CLI command options (--chunked-publish) (#13344)

* added Large Package Publishing e2e move test cases

* updated `large_packages.move` contract to support object code deployment and use SmartTable to store code in StagingArea

* added `--chunked-publish` option to the CLI for 1) account deploy 2) object deploy 3) object upgrade

* added `aptos move clear-staging-area` command

* added a new error type: `PackageSizeExceeded`

* updated CLI docs and output messages for the chunked publish mode

vusirikala committed:
Reduce frequency of calls to calculate remaining txns in proof queue (#14533)

sionescu committed:
Allow specifying the bucket for replay verify

sausagee committed:
[replay-verify] update ranges (#14548)

msmouse committed:
fix that DB::open_readonly not using the same cf options (#14572)

rahxephon89 committed:
[doc] support enum in docgen (#14562)

* doc for enum

* add tests

rahxephon89 committed:
temp (#14416)

vineethk committed:
[compiler-v2][lint] Needless ref deref pairs (#14581)

Adds a couple of lint checks to check and suggest removing needless ref-deref and deref-ref pairs.

larry-aptos committed:
Update the read me for indexer testing framework (#14549)

vusirikala committed:
Add counters in proof queue to measure garbage collection (#14532)

bchocho committed:
[mempool] improve parking lot eviction (#14586)

To better handle large transactions

heliuchuan committed:
Add support for a test federated oidc provider (#14580)

* Add support for a test federated oidc provider

* update

* fmt

sionescu committed:
[GHA] Apply input defaults to scheduled runs too

msmouse committed:
fix replay-verify: add quotes

aluon committed:
bump runs-on image to latest version (#14596)

igor-aptos committed:
calibrate forge (#14597)

Alexzander-Stone committed:
[framework] Fungible Asset - Mutable Metadata - Max length checks (#13917)

* Added check for max fields, similar to what `add_fungibility` does.

* FA tests for mutable metadata had too long of symbol length.

* FA tests for mutable metadata had too long of symbol length.

* Added new expecting-failure tests for each `mutate_metadata` parameter.

igor-aptos committed:
Have CoinInfo rust resources clarify their types (#14479)

JoshLind committed:
[Consensus Observer] Make subscription creation asynchronous.

igor-aptos committed:
test prologue speculation (#14575)

JoshLind committed:
[Consensus Observer] Move subscription health check to subscription.rs

JoshLind committed:
[Consensus Observer] Improve subscription manager unit tests.

vineethk committed:
[compiler-v2][lint] Extend the needless ref in field access lint checks to include enum variant field access (#14587)

perryjrandall committed:
[testing] Report unit test results to trunk.io

Test Plan: running on PR, step should upload results and not fail the job either way

rustielin committed:
[config] indexer grpc sanitizer prompts table_info_service_mode (#14613)

heliuchuan committed:
Add an entry function to update the federated jwkset (#14579)

* Add an entry function to update the federated jwkset

* make it an entry func

* update

* update md

* update func doc

* Update aptos-move/framework/aptos-framework/sources/jwks.move

Co-authored-by: Greg Nazario <greg@gnazar.io>

* update docs

---------

Co-authored-by: Greg Nazario <greg@gnazar.io>

larry-aptos committed:
[indexer][testing] Change to facilitate indexer localnode access. (#14550)

grao1991 committed:
Update base image for forge tests to v1.19.1 (#14617)

rahxephon89 committed:
[Compiler-v2] Fix bugs in index notation for vector type (#14609)

* fix dot exp for index notation

* move_pr.sh -i2 test

* add comment

gregnazario commented on aptos-move/framework/aptos-framework/sources/multisig_account.move:
```diff
@@ -547,6 +571,41 @@ module aptos_framework::multisig_account {
         );
     }
 
+    /// Private entry function that creates a new multisig account on top of an existing account and immediately rotate
+    /// the origin auth key to 0x0.
+    ///
+    /// Note: If the original account is a resource account, this does not revoke all control over it as if any
+    /// SignerCapability of the resource account still exists, it can still be used to generate the signer for the
+    /// account.
```
I would note here, also objects, don't prevent this as well.

yeptos replied:
specifically speaking things like ExtendRef can still exist for object account I assume that's what you're saying?

gregnazario commented on aptos-move/framework/aptos-framework/sources/multisig_account.move:
```diff
@@ -1688,6 +1747,26 @@ module aptos_framework::multisig_account {
     }
 
+    #[test]
+    public entry fun test_create_multisig_account_on_top_of_existing_with_signer()
```
tests don't have to be entry or public, guess it's kinda funny the ones in this file are set like that prior your changes.

yeptos replied:
for this pr i'll just go with the file's convention, think it's better for someone to change those as part of a refactoring commit

gregnazario reviewed (commented):
Looks fine to me

alinush committed:
add e2e move test for federated keyless (#14585)

* add e2e move test for federated keyless

* Use entry function instead of script to install JWKs

---------

Co-authored-by: Oliver <heliuchuan@gmail.com>

areshand committed:
add a field for internal indexer version

JoshLind committed:
[Consensus Observer] Improve subscription utility unit tests.

larry-aptos committed:
Create centralized config and transaction importer. (#14551)

grao1991 committed:
[Fix] Also enable event internal indexer on validator in forge. (#14534)

fEst1ck committed:
[Bugfix] [Compiler-V2] Fix a bug in control flow simplification (#14536)

* bug fix && doc && check invariants

* add tests

* format

* add exp file

* add test case

* typo

* add invariant

* refactor

* refactor

* update tests

* update tests

* doc

* comments

* add debug assert

* add invariant

* refactor

* add invariant

* format

* update tests

* update tests

* update tests

* correct doc

* bug fix

* linter

* address comments

* fix test case

* update tests

* update tests

---------

Co-authored-by: Zekun Wang <zekun.wang@aptoslabs.com>

vineethk committed:
[compiler-v2] Make flush writes optimization a default optimization (#14620)

zjma committed:
fed keyless smoke tests (#14359)

igor-aptos committed:
cleanup info logs (#14555)

igor-aptos committed:
cleanup warn logs (#14614)

igor-aptos committed:
remove backtrace from common warn logs (#14622)

gregnazario committed:
[cli] Release 4.2.0 (#14653)

gregnazario committed:
[cli] Add contribution guide (#14435)

JoshLind committed:
[Consensus Observer] Enable CO for VFNs.

wrwg committed:
[move-vm] Fixes to enum type implementation (#14657)

vusirikala committed:
Sync up QC in order vote message (#14637)

wrwg committed:
[compiler-v2] Disallow empty enum types (#14658)

Previously, those where compiled to empty structs which in turn lead to ZERO_STRUCT_SIZE bytecode verifier errors.

grao1991 committed:
Remove an unnecessary clone. (#14659)

0xbe1 committed:
fix `aptos move disassemble` help message (#14594)

JoshLind committed:
[Consensus Observer] Improve error messages for payload verification.

areshand committed:
use different metric name for each shard

vusirikala committed:
Cache commit votes received for future rounds in the buffer manager (#14570)

vgao1996 committed:
[gas] bump gas feature version to 1.20 (#14668)

vusirikala committed:
Deprecate delayed QC aggregate msg (#14640)

alinush committed:
timeless hotfix (#275)

sherry-x committed:
update forge version and gas version for release 1.21

rex1fernando committed:
Add comments to StringBodies circuit template (#14675)

larry-aptos committed:
update the localnet to use the latest processors. (#14682)

vineethk committed:
[compiler-v2][lint] Needless mutable reference (#14651)

brmataptos committed:
[move-compiler-v2] Add all remaining V1 tests to V2, except evm/async (#14578)

* add options and plumbing for compile_verify_code, warn_deprecated, etc., and add final set of V1 tests to V2, except flavor tests (evm/async)
* move around and edit/split tests to test same functionality in V2 as V1

gregnazario committed:
[cli] Release 4.2.1 (#14694)

Patches the indexer processors for 4.2.0

igor-aptos committed:
gas estimation changes (#14669)

brmataptos committed:
[move-compiler-v2] fix issue 6922 by allowing type annotations for lambda parameters (#14254)

Allowing declared type annotations for lambda parameters.

Fixes #6922.

vineethk committed:
[aptos-framework] Fix needless mutable references and borrows (#14656)

sionescu committed:
Update pin on the Swatinem/rust-cache GH action

The old version relies on a deprecated version of Node.js

sionescu committed:
[GHA] Update actions/checkout v3 -> v4

V3 runs on a deprecated version of Node.js (16) and Github has now
started forcing Node.js 20.

areshand committed:
skip version checking on get txn by hash

msmouse committed:
Fix potential deadlock in state_key::Entry::drop (#14670)

larry-aptos committed:
Fix the spamming log in CLI. (#14709)

areshand committed:
use storage ledgerinfo for txn APIs

areshand committed:
Revert "skip version checking on get txn by hash"

This reverts commit 1c65af50110fdd87e40070b4aebb8d7a7158907b.

areshand committed:
Revert "use storage ledgerinfo for txn APIs"

This reverts commit efe0365390bcc1d846329fa0dad9432174cad214.

grao1991 committed:
[Fix] Return None instead of error if table info is not available on the node. (#14715)

brmataptos committed:
[move-compiler-v2] Print a better error message for a string without b or x (#14687)

Fixes #14364.

msmouse committed:
[GHA] replay-verify jobs at snapshot boundaries (#14610)

brmataptos committed:
Add Move Compiler to components list in PR template, reorder sections to make more sense (#14384)

msmouse committed:
resolve rust-setup error

I put in the if- condition to speed up iteration on PRs, and it looks
like when the condition is not hit, rust-setup is not happy with it.
Removing to unbreak main.

```
Error:
aptos-labs/aptos-core/main/.github/actions/rust-setup/action.yaml: The
scalar style 'SingleQuoted' on line 23 and column 11 is not valid with
the tag '!startsWith(github.ref,'
Error: System.ArgumentException: Unexpected type '' encountered while
reading 'action manifest root'. The type 'MappingToken' was expected.
   at
   GitHub.DistributedTask.ObjectTemplating.Tokens.TemplateTokenExtensions.AssertMapping(TemplateToken
   value, String objectDescription)
      at
      GitHub.Runner.Worker.ActionManifestManager.Load(IExecutionContext
      executionContext, String manifestFile)
      Error: Failed to load
      aptos-labs/aptos-core/main/.github/actions/rust-setup/action.yaml
```

wrwg committed:
[compiler-v2][decompiler] Pulling fat-loop algorithm out of the prover (#14488)

The prover actually has a full-fledged loop analysis for stackless bytecode for loop invariant instrumentation and unrolling. This loop analysis is a major part of the logic for decompilation. This PR pulls out this functionality in a new general purpose module, `fat_loop`. Its mostly a refactoring of existing code.

grao1991 committed:
[Fix] Fix indexer FN and localnet. (#14727)

* [Fix] Only use version from internal indexer when it is enabled.

* fix the localnet support.

---------

Co-authored-by: Larry Liu <larry@aptoslabs.com>

msmouse committed:
Tune replay granularity for speed (#14729)

larry-aptos committed:
Release CLI 4.2.3. (#14731)

areshand committed:
sleep shorter and return 0 instead of throwing error

Commit:
[mempool] move the qs callback to be after mempool

msmouse committed:
[perf] optimize sort_transactions_with_state_checkpoint (#14748)

gregnazario committed:
[ts-sdk] Remove V1 SDK (#14700)

* [ts-sdk] Remove V1 SDK

* [CI] Remove TS SDK E2E tests

* [api] Update readme to not refer to legacy TS SDK

* [refactor] Remove ts-sdk more

* [refactor] Deprecate rest of SDK v1

Rqnsom committed:
[aptos-cli]: update move-tool visibility (#14735)

Ensure that the following functions are publicly visible so they can be
used from the `move-mutation-test` tool.
- `aptos::move_tool::aptos_debug_natives::aptos_debug_natives()`
- `aptos::move_tool::experiments_from_opt_level()`

sherry-x committed:
Update mod.rs

yzaccc committed:
Automatically bump Forge compat base image to latest release (#14686)

* Automatically bump Forge compat base image to latest release

---------

yzaccc committed:
adding new runs on image for testing, included gcloud sdk and kubectl (#14765)

Co-authored-by: Zac Yang <zac.yang@aptoslabs.com>

vusirikala committed:
LedgerInfoWithUnverifiedSignatures data structure to store both verified and unverified signatures. (#14641)

rahxephon89 committed:
[Compiler-v2] Check recursive definition for constants (#14741)

* check recursive in constant definition

* add tests

* add tests for duplicated constants

* fix

heliuchuan committed:
Add Cognito support to the pepper service (#14761)

yzaccc committed:
added gcloud sdk and kubectl in the packer, uploaded the image in to … (#14763)

* added gcloud sdk and kubectl in the packer, uploaded the image in to ci aws account, changed the ami in workflows

---------
Co-authored-by: Zac Yang <zac.yang@aptoslabs.com>

rustielin committed:
[gha] fix cargo metadata syntax error (#14772)

* [gha] fix cargo metadata syntax error

* [gha] cargo metadata uploader uses gcloud storage

JoshLind committed:
[Consensus Observer] Skip expired batches from verification.

areshand committed:
cargo machete remove

JoshLind committed:
[Dev Setup] Pin cargo-machete version

yeptos committed:
add an integration test

yeptos committed:
fix arg names

github-actions[bot] committed:
Update Docker images (#14798)

Co-authored-by: gedigi <19227040+gedigi@users.noreply.github.com>

Commit:
[mailmap] Remove mailmap (#14803)

rahxephon89 committed:
format `tests` and `examples` folder (#14779)

igor-aptos committed:
recalibrate single node performance (mostly minor increases) (#14804)

larry-aptos committed:
[EI-519][indexer][testing] Implement scripted transaction generation (#14569)

* script transaction

* modify the output path by append the network name

---------

Co-authored-by: yuunlimm <dla0267@gmail.com>

yeptos committed:
fix lint
